core/src/main/java/io/methvin/watcher/DirectoryWatcher.java

```diff
@@ -198,8 +239,15 @@ public void watch() {
             throw new IllegalStateException(
                 "WatchService returned key [" + key + "] but it was not found in keyRoots!");
           }
+
           Path childPath = eventPath == null ? null : keyRoots.get(key).resolve(eventPath);
-          logger.debug("{} [{}]", kind, childPath);
+          if (shouldIgnoreEvent(childPath)) {
```

Comment:
I don't understand why we need this. My assumption is that `register(_, false)` will register a directory in a non-recursive way. Is that not the case? If it isn't, can we disable recursive file watching at all for these directories in particular so that we don't need to pollute the logic of the `watch` method?

Reply:
`register(_, false)` technically should watch non-recursively, but currently the macOS WatchService implementation always watches recursively regardless of what modifier is passed. That's because originally the goal of this library was *only* to watch directories recursively. One solution is to add this same filtering logic to the `MacOSXListeningWatchService`, since that's the only place it should be needed.